src/handlers/dns/DnsStreamHandler.cpp

```diff
@@ -335,14 +334,18 @@ bool DnsStreamHandler::_filtering(DnsLayer &payload, [[maybe_unused]] PacketDire
         std::string_view qname_ci = payload.getFirstQuery()->getNameLower();
         for (const auto &fqn : _f_qnames) {
             // if it matched, we know we are not filtering
-            if (endsWith(qname_ci, fqn)) {
+            if (ends_with(qname_ci, fqn)) {
                 _static_suffix_size = fqn.size();
                 goto will_not_filter;
             }
         }
         // checked the whole list and none of them matched: filter
         goto will_filter;
     }
+    // This is not actually a filter, but only should work if OnlyQNameSuffix is not enabled
+    else if (_f_enabled[Filters::PublicSuffixList] && payload.parseResources(true) && payload.getFirstQuery() != nullptr) {
```

Comment:
did you use _f_enabled for convenience? i would rather not conflate it with config options which are not filters. if it makes sense we can add _c_enabled and a similar list of config functionality

Reply:
done